File: backend/src/main/java/com/gymflow/pro/entity/Workout.java
package com.gymflow.pro.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "workouts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Workout {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id", nullable = false)
    private Student student;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "instructor_id")
    private User instructor;

    @Column(nullable = false, length = 50)
    private String name;

    private String goal;

    @Builder.Default
    @Column(nullable = false)
    private boolean active = true;

    @Builder.Default
    @Column(name = "start_date", nullable = false)
    private LocalDate startDate = LocalDate.now();

    @Column(name = "end_date")
    private LocalDate endDate;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Builder.Default
    @OneToMany(mappedBy = "workout", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("sortOrder ASC")
    private List<WorkoutExercise> exercises = new ArrayList<>();

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: frontend/src/types/workout.ts
export interface WorkoutExercise {
  id: string
  exerciseId: string
  exerciseName: string
  sets: number
  reps: string
  load?: string | number
  restSeconds?: number
  durationSeconds?: number
  order: number
  notes?: string
}

export interface Workout {
  id: string
  studentId: string
  studentName: string
  name: string
  goal?: string
  active: boolean
  startDate?: string
  endDate?: string
  notes?: string
  exercises: WorkoutExercise[]
  createdAt: string
  updatedAt: string
}

export interface WorkoutExercisePayload {
  exerciseId: string
  sets: number
  reps: string
  load?: string
  restSeconds: number
  order: number
}

export interface WorkoutPayload {
  studentId: string
  name: string
  goal?: string
  active: boolean
  startDate?: string
  endDate?: string
  notes?: string
  exercises: WorkoutExercisePayload[]
}
